### DISCUSSION TITLE
why we don't like Next.js

### DISCUSSION THREAD

**Discussion by tom2drum** [blockscout/frontend#167] [General]
Let's discuss what Next.js cannot do or doing in the way what is not convenient for our needs

**Comment by isstuev** [comment]
nextjs can't match routes with unknown number of subfolders
we need to match '/<network_type>/txs',  '/<network_type>/<network_subtype>/txs', '/<network_type>/<network_subtype>/.../txs' to the same page and have to use rewrites for each level

**Comment by tom2drum** [comment]
out of the box nextjs app cannot be configure at runtime, because all environment variables are injected at build time and cannot be changes later

currently we use a workaround for this but with some minor limitations
